index.html

```diff
@@ -3510,35 +3509,60 @@ <h1 id="title" class="text-xl font-bold mx-auto truncate w-full">
             ></path>
           </svg>
         </button>
+
+        <button
+          id="toggle_chat"
+          onClick="toggleAppMode(); render();"
+          class="rounded-md border border-transparent bg-slate-600 py-2 px-2 text-sm font-medium text-white shadow-sm hover:bg-slate-700 focus:outline-none focus:ring-2 focus:ring-slate-500 focus:ring-offset-2 min-w-[38px]"
+        >
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            fill="none"
+            viewBox="0 0 24 24"
+            stroke-width="1.5"
+            stroke="currentColor"
+            class="h-5 w-5"
+          >
+            <path
+              stroke-linecap="round"
+              stroke-linejoin="round"
+              d="M7.5 8.25h9m-9 3H12m-9.75 1.51c0 1.6 1.123 2.994 2.707 3.227 1.129.166 2.27.293 3.423.379.35.026.67.21.865.501L12 21l2.755-4.133a1.14 1.14 0 01.865-.501 48.172 48.172 0 003.423-.379c1.584-.233 2.707-1.626 2.707-3.228V6.741c0-1.602-1.123-2.995-2.707-3.228A48.394 48.394 0 0012 3c-2.392 0-4.744.175-7.043.513C3.373 3.746 2.25 5.14 2.25 6.741v6.018z"
+            />
+          </svg>
+        </button>
       </div>
     </nav>
 
     <main
-      class="h-full flex justify-between portrait:flex-col landscape:flex-row gap-4 px-4 py-2"
+      class="h-[calc(100%-54px)] flex flex-col gap-4 px-4 py-2 justify-between portrait:flex-col landscape:flex-row"
       x-data="{ ...messageData }"
+      id="main"
     >
       <!-- Start game -->
       <canvas
-        class="block aspect-square portrait:fixed portrait:w-screen portrait:left-0 landscape:h-[70%]"
         id="canvas"
+        class="block fixed left-0"
         width="400"
         height="400"
       ></canvas>
       <!-- End game -->
 
       <!-- container for both chat text and message box -->
-      <div class="flex flex-col landscape:h-[70%] landscape:grow">
+      <div class="flex flex-col landscape:h-[70%] landscape:grow h-full">
         <!-- Start chat/rooms list -->
         <!-- flex-col-reverse makes div scroll to bottom by default -->
         <div
-          class="flex flex-col-reverse box-border"
+          class="flex flex-col-reverse box-border h-full"
```

Comment:
i was wondering why the 2 `h-fulls` here dont push the reply box below the screen, since this would mean the container div takes up the same height as `<main>`, and then i saw the extremely cursed `h-[calc(100%-54px)]` class on `<main>` which means `<footer>` is indeed getting pushed below out of `<main>` but since there's space below, it remains on screen.

Wait doesn't that `h-[calc(100%-54px)]` class get overridden by `viewportHax.js::setViewHeight()` anyway? It isn't doing anything, is it?

Reply:
i'm not sure, does setViewHeight run on init? Honestly last night i was running on fumes, i tried to remove that cursed h-calc class and it broke, but doing it now might work because lots more other changes have been made / undone

Reply:
> does setViewHeight run on init?

yep, on line 13 in viewportHax.js

> i tried to remove that cursed h-calc class and it broke

it will probably still break now unless we change `#view` from `h-full` to smth else. We could do something like `#view` 90%, `<footer>` 10%. Any suggestions @Johnnycus ?

Reply:
yeah i guess we can try